src/checkers/complexity.js

```diff
@@ -0,0 +1,141 @@
+import React from "react";
+import text from "../words_alpha.json"
+
+const formatFeedback = feedback => {
+//	if (!feedback.warning && !feedback.suggestions) {
+//		return <p></p>
+//	}
+	return (
+		<>
+			<>
+				<h4>Suggestions:</h4>
+				<ul>
+					<p>{feedback}</p>
+				</ul>
+			</>
+		</>
+	);
+};
+
+const determineStrength = password => {
+	// w is password
+	// n is number of charsets in w
+	// k is the number of same-charset-substrings
+	// l is the length of password
+	// s is a bonus if w has special characters
+	// p is special character position penalty
+	// d/l is the in-dictionary penalty.
+	// s = 0.5 if w contains a special character, 0 otherwise
+	// p = 0.5 if special character is at the beginning or the end of w and w has no more 
+	// 			than 2 charsets. 0 otherwise
+	// d is the length of the substring that is a dictionary word
+	var special = "*|,\":<>[]{}`';()@&$#%";
+	var num = "1234567890";
+	var word = "abcdefghijklmnopqrstuvwzyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	var l = password.length;
+	if (l < 4) {
+		var feedback = "Password needs to be longer than 4 characters";
+		return {strength: 0, description: formatFeedback(feedback)}
+	}
+	var n = 0.0; // this is because we have the default charset = utf-8 we are gonna assume thats 1
+	var k = 0.0; // all of the substrings are in the same charset
+	var wordsub = [];
+	var numsub = [];
+	var specialsub = [];
+	var numstr = "";
+	var specialstr = "";
+	var wordstr = "";
+	var s = 0.0;
+	var p = 0.0;
+	var specialc,numc,wordc = 0;
+	for (var e = 0; e < l; e++){
+		if (num.indexOf(password.charAt(e)) !== -1){
```

Comment:
I'd recommend using `num.includes` here instead of `num.indexOf` and in the other checks, slightly more readable imo